Dockerfile.s390x

```
@@ -0,0 +1,168 @@
+# Base UBI image for s390x architecture
+ARG BASE_UBI_IMAGE_TAG=9.5-1736404155
+ARG PYTHON_VERSION=3.12
+FROM registry.access.redhat.com/ubi9/ubi-minimal:${BASE_UBI_IMAGE_TAG} AS base
+
+# Install basic dependencies
+ARG PYTHON_VERSION
+ENV PYTHON_VERSION=${PYTHON_VERSION}
+RUN microdnf -y update && microdnf install -y \
+    python${PYTHON_VERSION}-pip python${PYTHON_VERSION}-wheel \
+    && microdnf clean all
+
+WORKDIR /workspace
+
+ENV LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8
+
+# Install development utilities
+RUN microdnf install -y \
+    which procps findutils tar vim git gcc g++ make patch make cython zlib-devel \
+    libjpeg-turbo-devel libtiff-devel libpng-devel libwebp-devel freetype-devel harfbuzz-devel \
+    openssl-devel openblas openblas-devel wget autoconf automake libtool cmake && \
+    microdnf clean all
```

Installing `cython` causes python3.9 to be installed, this should be updated like so:

```suggestion
RUN microdnf install -y \
    which procps findutils tar vim git gcc g++ make patch make python${PYTHON_VERSION}-Cython zlib-devel \
    libjpeg-turbo-devel libtiff-devel libpng-devel libwebp-devel freetype-devel harfbuzz-devel \
    openssl-devel openblas openblas-devel wget autoconf automake libtool cmake && \
    microdnf clean all
```

---

Looking at the [pyarrow requirements](https://github.com/apache/arrow/blob/main/python/requirements-build.txt#L1C1-L1C9), we don't really need to iinstall `cython` from the system package manager, since we're already installing `requirements-build.txt` bellow in the pyarrow build stage.